### ISSUE TITLE
Proposal: add exclusion-checks

### ISSUE THREAD

**Issue by funkyfuture** [nicolaiarocci/cerberus#84]
it'd be handy to declare properties to exlude others:

```python
{'property_a': {'excludes': 'property_b'},
 'property_b': {'excludes': ['property_a', 'property_c']}
```

if a property is not present in the document, excluded by a present one, but marked required, the requirement should be dismissed. this way mutual exclusive requirements are possible:

```python
schema = \
{'property_a': {'excludes': 'property_b', 'required': True},
 'property_b': {'excludes': 'property_a', 'required': True}}

valid_documents = \
[{'property_a': 'foo'}, {'property_c': 'bar'}]

invalid_documents = \
[{'property_a': 'foo', 'property_b': 'bar'}, {'property_c': 'baz'}]
```

since i can make use of this and may be implementing the next days, i'd appreciate any thoughts on this.

**Comment by nicolaiarocci** [comment]
Looks good to me!

**Comment by funkyfuture** [comment]
eventually it turned out, that this wouldn't be of help for me for now.

so, maybe later, but certainly not within the next days. so anyone who feels the urge, feel free to pick it. shouldn't be to tricky either.

**Comment by funkyfuture** [comment]
wouldn't it be a good idea to keep this issue open?

**Comment by funkyfuture** [comment]
i have the *feeling* that my use-case may be covered with the *of-rules. however i still like this more obvious way.

**Comment by funkyfuture** [comment]
solved by #137. :rocket:
